### PR TITLE
Finalize Ruby support

### PR DESCRIPTION
**Pull request by ruiAzevedo19**
Part of #300

Early Merger is #335

### CODE REVIEW

File: language/ruby/language.go [hunk 1]
```diff
@@ -57,11 +64,65 @@ func (l *Language) DefaultTestFileSuffix() string {
...
+
+	testResult.Coverage, err = language.CoverageObjectCountOfFile(logger, coverageFilePath)
+	if err != nil {
+		return nil, nil, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+	}
 
 	return testResult, problems, nil
 }
 
+var symflowerTestSummaryRe = regexp.MustCompile(`(\d+) runs, \d+ assertions, (\d+) failures, (\d+) errors, \d+ skips`)
```
Comment (bauersimon) [inline language/ruby/language.go:102]: `languageRakeTestSummaryRE`

Comment (bauersimon) [reply language/ruby/language.go:102]: Also not sure if we would want to subtract the `skips`?

File: language/ruby/language_test.go [hunk 2]
```diff
@@ -62,3 +64,75 @@ func TestLanguageImportPath(t *testing.T) {
...
+			TestBalancedBrackets#test_has_balanced_brackets5:
+			NoMethodError: undefined method ` + "`" + `has_balancedbrackets' for an instance of TestBalancedBrackets
+    			test/balanced_brackets_test.rb:31:in ` + "`" + `test_has_balanced_brackets5'
+
+			6 runs, 5 assertions, 1 failures, 1 errors, 0 skips
+			rake aborted!
+			Command failed with status (1)
+		`,
+		ExpectedTestsTotal: 6,
+		ExpectedTestsPass:  4,
```
Comment (bauersimon) [inline language/ruby/language_test.go:117]: Why are there 5 assertions if two of the tests failed? I guess because failed assertions also count?

**Review by bauersimon — DISMISSED** [review]

File: language/ruby/language.go [hunk 3]
```diff
@@ -65,11 +71,120 @@ func (l *Language) DefaultTestFileSuffix() string {
...
+		}
+		filePath := filepath.Join(repositoryPath, "test", entry.Name())
+
+		fileContent, err := os.ReadFile(filePath)
+		if err != nil {
+			return pkgerrors.WithStack(err)
+		}
+
+		// Only add the import if it does not exist already.
+		if strings.Contains(string(fileContent), "require_relative \"test_init\"") {
```
Comment (Munsio) [inline language/ruby/language.go:162]: use ```

File: testdata/ruby/light/repository.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "tasks": ["write-tests"]
```
Comment (Munsio) [inline testdata/ruby/light/repository.json:2]: wrong formatting

File: Dockerfile
Comment (bauersimon) [inline Dockerfile:1]: This is a temporary commit that should be removed before we merge this.

File: language/ruby/language.go [hunk 5]
```diff
@@ -65,11 +71,120 @@ func (l *Language) DefaultTestFileSuffix() string {
...
+	if err != nil {
+		return 0, 0, pkgerrors.WithStack(err)
+	}
+
+	// The "assertions" cannot be used to calculate the passing tests, cause they also include failed assertions.
+	return testsTotal, testsTotal - testsFailure - testsErrors - testsSkips, nil
+}
+
+// injectCoverageTracking injects our custom coverage tracking logic into a ruby project.
+func injectCoverageTracking(repositoryPath string) error {
```
[the remark this answers is not in the document]
Comment (bauersimon) [reply language/ruby/language.go:144]: Basically whenever an LLM writes a test yes... But it is done always. However there are checks such that it works with existing files.

File: language/ruby/language.go (hunk 3 above)
Comment (bauersimon) [reply language/ruby/language.go:162]: I have the same thing below but with a `\n`, then the backticks don't work anymore. Would rather keep it consistent.

File: README.md [hunk 6]
```diff
@@ -142,6 +142,7 @@ The execution by default also creates a report file `REPORT.md` that contains ad
 The following parameters do have a special behavior when using a containerized runtime.
 
 - `--testdata`: The check if the path exists is ignored on the host system but still enforced inside the container because the paths of the host and inside the container might differ.
+- `--configuration`: Passing configuration files to the docker runtime is currently unsupported.
```
Comment (zimmski) [inline README.md:145]: Sort

**Review by Munsio — APPROVED** [review]

**Comment by Munsio** [comment]
@bauersimon only a nitpick left which we can ignore. Good Job you two to get this up and running.

**Pull request merged**